transaction/core/src/tx_summary.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! The TxSummary is meant to reduce the complexity of implementing a hardware
+//! wallet. (See MCIP 52 also.)
+//!
+//! Hardware wallets have the following issue:
+//! * The main function they want to perform is signing RingMLSAG. This only
+//!   requires sending about 1kb of data and can be fast even on a slow
+//!   connection and slow device.
+//! * However, if they want to be able to display to the user what it is that
+//!   they are signing, i.e. how much money is being transfered to who, and
+//!   verify that this is correct on the device, then they have to be able to
+//!   trace the "extended message digest" back to the TxPrefix.outputs list and
+//!   see where these inputs they are signing away are going to.
+//! * Unfortunately, computing the extended message digest is very annoying,
+//!   because it depends on essentially the entire TxPrefix. A Tx may have as
+//!   many as 16 TxIn's. Each TxIn has 11 mixins. Each mixin may contain a
+//!   merkle proof with say 30 merkle proof elements. Each merkle proof element
+//!   is 40 bytes. So suddenly, we have a need to stream > 100 kb of data to the
+//!   device so that it can compute this digest, and if any piece of data is
+//!   skipped, then the digest will be wrong and cannot be verified.
+//!
+//! The point of the TxSummary is to avoid the need for them to have to stream
+//! all of this data to be confident about what is the balance of the Tx that
+//! they are signing.
+//!
+//! The idea is similar to merkle proof verification. You don't have to see the
+//! entire data set to be convinced that the piece you care about is part of the
+//! hash, you can be supplied with hashes for the branches of the tree that you
+//! aren't specifically interested in, and then be convinced that the data you
+//! are interested is part of the root hash.
+//!
+//! So, we:
+//! * Take the existing extended message digest (32 bytes)
+//! * Use that to start a new Merlin transcript
+//! * Martial the TxSummary into this new Merlin transcript
+//! * A 32 byte digest resulting from that is new "extended message with tx
+//!   summary digest", and this is what the MLSAG actually sign.
+//!
+//! We propose that the TxSummary contains:
```

Comment:
Nit: "We propose" seems out of place in a code comment
```suggestion
//! The TxSummary contains:
```